### ISSUE TITLE
formatOnSave doesnt always format

### ISSUE THREAD

**Issue by jeromeetienne** [bichiliad/atom-jsfmt#8]
It seems the format action may be taken *after* the save

I fixed it by changing the 
```
    editor.getBuffer().on 'saved'
```

into a 

```
    editor.getBuffer().onWillSave =>
```

**Comment by jeromeetienne** [comment]
additionally 'saved' is deprecated. see below

https://github.com/atom/text-buffer/blob/master/src/text-buffer.coffee#L279

```
      when 'will-be-saved'
        Grim.deprecate("Use TextBuffer::onWillSave instead. A TextBuffer instance is no longer provided as a callback argument.")
      when 'saved'
        Grim.deprecate("Use TextBuffer::onDidSave instead. A TextBuffer instance is no longer provided as a callback argument.")
      when 'will-reload'
        Grim.deprecate("Use TextBuffer::onWillReload instead.")
```

**Comment by bichiliad** [comment]
I believe this is fixed as of 0.7.1. Thanks for bringing it up! I'm still working on a way to test this sanely.
